File: internal/content/testdata/bench/go/sort-array-of-three-values.go
func sort_three_values(arr []int, low_val int, mid_val int, high_val int) []int {
	lo, mid, hi := 0, 0, len(arr)-1
	for mid <= hi {
		if arr[mid] == low_val {
			arr[lo], arr[mid] = arr[mid], arr[lo]
			lo++
			mid++
		} else if arr[mid] == mid_val {
			mid++
		} else {
			arr[mid], arr[hi] = arr[hi], arr[mid]
			hi--
		}
	}
	return arr
}

File: internal/content/testdata/bench/rust/sort-array-of-three-values.rs
fn sort_three_values(mut arr: Vec<i64>, low_val: i64, mid_val: i64, high_val: i64) -> Vec<i64> {
    let _ = high_val;
    let mut lo = 0usize;
    let mut mid = 0usize;
    let mut hi = arr.len() as i64 - 1;
    while mid as i64 <= hi {
        if arr[mid] == low_val {
            arr.swap(lo, mid);
            lo += 1;
            mid += 1;
        } else if arr[mid] == mid_val {
            mid += 1;
        } else {
            arr.swap(mid, hi as usize);
            hi -= 1;
        }
    }
    arr
}
